src/modal-manager-controller.ts

```diff
@@ -0,0 +1,138 @@
+import './modal-manager';
+import { ModalConfig } from './modal-config';
+
+/**
+ * The ModalManagerController is a bridge between the host app and the ModalManager
+ *
+ * It is responsible for handling environmental interactions such as setting up
+ * window resize listeners to account for resizing, as well as setting CSS
+ * properties on the document to disable scrolling when open.
+ *
+ * This may not work for all environments but can be used
+ * as a base class to extend for specific needs.
+ *
+ * @export
+ * @class ModalManagerController
+ * @extends {LitElement}
+ * @implements {ModalManagerInterface}
+ */
+@customElement('modal-manager-controller')
+export class ModalManagerController extends LitElement
+  implements ModalManagerInterface {
+  /**
+   * Custom content to inject into the Modal Manager for passing through slotted content
+   *
+   * @type {TemplateResult}
+   * @memberof ModalManagerController
+   */
+  @property({ type: Object }) slotContent?: TemplateResult;
+
+  /** Reference to the ModalManager DOM element */
+  @query('modal-manager') private modalManager!: ModalManagerInterface;
+
+  /** @inheritdoc */
+  render(): TemplateResult {
+    return html`
+      <modal-manager @modeChanged=${this.modalModeChanged}>
+        ${this.slotContent}
+      </modal-manager>
+    `;
+  }
+
+  /** @inheritdoc */
+  getMode(): ModalManagerMode {
+    return this.modalManager.getMode();
+  }
+
+  /**
+   * Show a modal with a given configuration.
+   *
+   * @param {{
+   *     config: ModalConfig;
+   *     customModalContent?: TemplateResult;
+   *     userClosedModalCallback?: () => void;
+   *   }} options
+   * @returns {Promise<void>}
+   * @memberof ModalManagerController
+   */
+  async showModal(options: {
+    config: ModalConfig;
+    customModalContent?: TemplateResult;
+    userClosedModalCallback?: () => void;
+  }): Promise<void> {
+    return this.modalManager.showModal(options);
+  }
+
+  /**
+   * Close the modal
+   *
+   * @returns {void}
+   * @memberof ModalManagerController
+   */
+  closeModal(): void {
+    return this.modalManager.closeModal();
+  }
+
+  /** @inheritdoc */
+  firstUpdated(): void {
+    window.addEventListener('resize', () => {
+      this.modalManager.style.setProperty(
+        '--containerHeight',
+        `${window.innerHeight}px`
+      );
+    });
+
+    this.modalManager.style.setProperty(
+      '--containerHeight',
+      `${window.innerHeight}px`
+    );
+  }
+
+  /**
+   * Callback that handles when the ModalManager changes modes.
+   *
+   * This is where we do things like stop the document body from scrolling.
+   *
+   * @private
+   * @param {CustomEvent} e
+   * @memberof ModalManagerController
+   */
+  private modalModeChanged(e: CustomEvent): void {
+    const mode = e.detail.mode as ModalManagerMode;
+    switch (mode) {
+      case ModalManagerMode.Modal:
+        document.body.style.overflow = 'hidden';
```

Comment:
**Suggestion**: toggle a class on the body rather than directly set a CSS property, especially since the default value for overflow is "visible" rather than "auto". Then the parent document can control what happens in these two states.

Reply:
This is where things were getting tricky regarding responsibilities between the host and the modal manager itself.

Originally I had all of this logic in petabox so as to let it handle things like overflow, but there's a decent amount of interaction that has to happen for it to all work properly that I didn't want to saddle on the host.

I instead created this `<modal-manager-controller>`, which is a somewhat generic bridge between the host and the actual `<modal-manager>`, which it wraps and passes through. It makes a few assumptions like this one to require as little integration work on the consumer as possible.

Consumers could take the `<modal-manager>` directly and handle this logic without using the controller or even subclass the controller to handle their environment.

I might be able to just pull out a few of the environmental-specific things like this. I need to think of how to do that.

Reply:
I switched it to use a class so the consumer just has to add
```
body.modal-manager-open {
  overflow: hidden;
}
```
and it'll add/remove that selector.

Reply:
So I ended up dropping the `modal-manager-controller` class in favor of a sidecar class called the `ModalManagerHostBridge`. It is responsible for handling these host interactions and we provide a default class, which can be overridden by the consumer if they want to do things differently. It also avoids the awkward wrapping that the controller was causing.